st2common/st2common/util/misc.py

```diff
@@ -92,3 +93,18 @@ def rstrip_last_char(input_str, char_to_strip):
         return input_str[:-len(char_to_strip)]
 
     return input_str
+
+
+def deep_update(d, u):
```

Comment:
I was looking for an existing library, but the one I found on PyPi actually breaks on some edge cases so I just decided to use a simple function.